include/rmm/mr/device/fixed_size_memory_resource.hpp

```diff
@@ -121,11 +120,11 @@ class fixed_size_memory_resource
   [[nodiscard]] std::size_t get_block_size() const noexcept { return block_size_; }
 
  protected:
-  using free_list  = detail::fixed_size_free_list;
-  using block_type = free_list::block_type;
+  using free_list  = detail::fixed_size_free_list;  ///< The free list type
+  using block_type = free_list::block_type;         ///< The type of block managed by the free list
   using typename detail::stream_ordered_memory_resource<fixed_size_memory_resource<Upstream>,
                                                         detail::fixed_size_free_list>::split_block;
-  using lock_guard = std::lock_guard<std::mutex>;
+  using lock_guard = std::lock_guard<std::mutex>;  ///< RAII lock guard
```

Comment:
Why do we have to document protected definitions?

Reply:
Protected members (at least of non-final classes) are still part of the API in the sense that someone could subclass your class and need them, right?

Reply:
OK. But the type of lock is an implementation detail. Perhaps it should be `private`.

Reply:
Yes, that would be a reasonable change. I can open an issue to track that change (would rather not lump functional changes into this PR).

Reply:
https://github.com/rapidsai/rmm/issues/1331